esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -431,7 +432,8 @@ void EsdlServiceImpl::configureUrlMethod(const char *method, IPropertyTree &entr
     entry.setProp("@prot", protocol);
     entry.setProp("@path", path);
 
-    Owned<ISmartSocketFactory> sf = createSmartSocketFactory(iplist, true);
+    Owned<ISmartSocketFactory> sf = new CSmartSocketFactory(iplist, true);
```

Comment:
There wasn't an actual need to follow the factory pattern here (although factories can be very elegant in some cases). We need an instance of SmartSocket, the create method currently only instantiates a SmartSocket, in the future it could instantiate others.